### ISSUE TITLE
meta.source field in subscription messages is inconsistently populated

### ISSUE THREAD

**Issue by Qingyixia** [hapifhir/hapi-fhir#4523]
**Describe the bug**
We have FHIR subscriptions that send all resource updates to Kafka. We have noticed that the {{meta.source}} field in the messages is sometimes present and sometimes missing. Furthermore, we have found that in some cases the {{meta.source}} field is actually the {{meta.source}} from the previous version of the resource. The behavior of this field depends on the request method. This means we can’t reliably correlate messages in Kafka with log messages in other systems. Ideally, this field would behave consistently in all messages.

**To Reproduce**
1. Create a message Subscription
2. Create a Patient resource
3. Inspect the Kafka topic. The latest message will be a CREATE message that has no `meta.source` field.
4. Update the Patient with a PUT request
5. Inspect the Kafka topic. The latest message will be an UPDATE message that has no `meta.source` field.
6. Update the Patient resource with a PATCH request
7. Inspect the Kafka topic. The latest message will be an UPDATE message *with* a `meta.source` field but it's the previous version
To summarize:
* `meta.source` is missing when the request is a POST or PUT
* `meta.source` is populated with the previous version when the request is PATCH

**Expected behavior**
meta.source field would behave consistently in all messages.
